zcash_client_sqlite/src/wallet/init/migrations/full_account_ids.rs

```diff
@@ -110,17 +112,28 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
                         })?,
                     )
                     .map_err(|_| {
-                        WalletMigrationError::CorruptedData(
-                            "Unable to derive spending key from seed.".to_string(),
-                        )
+                        if seed_is_relevant {
+                            WalletMigrationError::CorruptedData(
+                                "Unable to derive spending key from seed.".to_string(),
+                            )
+                        } else {
+                            WalletMigrationError::SeedNotRelevant
```

Comment:
I'm confused by your comment, because what you describe is precisely what I'm checking. This is the migration variant of the bug that I fixed in `WalletRead::validate_seed`:
- If we encounter this error with the _first_ derived account we're checking, we can assert that the seed is not relevant to the wallet by assuming 🅰️ that all derived accounts are from the same seed, and 🅱️ that all derived accounts in the wallet _must_ have been able to derive their USKs (in order to derive UIVKs).
- Once the seed has been determined to be relevant (because it matched the first derived account), any _subsequent_ account derivation failure is proving our above assumption 🅱️ wrong, and hence (per our discussion in our pairing where we are preserving assumption 🅰️ for Zashi 1.0) we report this as corrupted data like the migrations previously did.

Reply:
It's the fact that order is relevant to which error is returned that tripped me up here, and that is confusing when reading the code. A comment to indicate why the error type depends upon the order would be helpful.